From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NONPLAYT <nonplay@bxteam.org>
Date: Sun, 2 Feb 2025 14:43:32 +0300
Subject: [PATCH] Option to disable method profiler


diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index d483bcd450dcfe9d58cde804030ded570af6e25a..1b4947484a01480d818a886bcd1d8a647cb71aef 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -2810,6 +2810,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
     // CraftBukkit end
 
     private ProfilerFiller createProfiler() {
+        if (org.bxteam.divinemc.config.DivineConfig.PerformanceCategory.disableMethodProfiler) return net.minecraft.util.profiling.InactiveProfiler.INSTANCE; // DivineMC - Option to disable method profiler
         if (this.willStartRecordingMetrics) {
             this.metricsRecorder = ActiveMetricsRecorder.createStarted(
                 new ServerMetricsSamplersProvider(Util.timeSource, this.isDedicatedServer()),
