From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NONPLAYT <76615486+NONPLAYT@users.noreply.github.com>
Date: Sun, 6 Jul 2025 02:10:59 +0300
Subject: [PATCH] SparklyPaper: Clear dirty stats after copying

Original project: https://github.com/SparklyPower/SparklyPaper

diff --git a/net/minecraft/stats/ServerStatsCounter.java b/net/minecraft/stats/ServerStatsCounter.java
index 16e06085ed3c0f6a0b35c730b63b75824f44a905..b2d8f5d146ec10d34a4e71cbfc975fb68e5a8d04 100644
--- a/net/minecraft/stats/ServerStatsCounter.java
+++ b/net/minecraft/stats/ServerStatsCounter.java
@@ -98,12 +98,6 @@ public class ServerStatsCounter extends StatsCounter {
         this.dirty.add(stat);
     }
 
-    private Set<Stat<?>> getDirty() {
-        Set<Stat<?>> set = Sets.newHashSet(this.dirty);
-        this.dirty.clear();
-        return set;
-    }
-
     public void parseLocal(DataFixer fixerUpper, String json) {
         try {
             JsonElement jsonElement = StrictJsonParser.parse(json);
@@ -139,10 +133,12 @@ public class ServerStatsCounter extends StatsCounter {
     public void sendStats(ServerPlayer player) {
         Object2IntMap<Stat<?>> map = new Object2IntOpenHashMap<>();
 
-        for (Stat<?> stat : this.getDirty()) {
+        for (Stat<?> stat : this.dirty) { // DivineMC - SparklyPaper: Skip dirty stats copy when requesting player stats
             map.put(stat, this.getValue(stat));
         }
 
+        this.dirty.clear(); // DivineMC - SparklyPaper: Clear dirty stats after copying
+
         player.connection.send(new ClientboundAwardStatsPacket(map));
     }
 }
