From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NONPLAYT <nonplay@bxteam.org>
Date: Sun, 23 Feb 2025 01:14:54 +0300
Subject: [PATCH] Catch update suppressors


diff --git a/net/minecraft/network/PacketProcessor.java b/net/minecraft/network/PacketProcessor.java
index 408cb996fc8ff77785da8919f19baff2cf4bd789..f398d48910dff89c41d1344ebf7b793c27dcb6aa 100644
--- a/net/minecraft/network/PacketProcessor.java
+++ b/net/minecraft/network/PacketProcessor.java
@@ -95,6 +95,10 @@ public class PacketProcessor implements AutoCloseable {
             if (this.listener.shouldHandleMessage(this.packet)) {
                 try {
                     this.packet.handle(this.listener);
+                // DivineMC start - Catch update suppressors
+                } catch (org.bxteam.divinemc.util.exception.UpdateSuppressorException e) {
+                    LOGGER.info(e.getMessage());
+                // DivineMC end - Catch update suppressors
                 } catch (Exception e) {
                     if (e instanceof ReportedException re && re.getCause() instanceof OutOfMemoryError) {
                         throw PacketUtils.makeReportedException(e, this.packet, this.listener);
diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index afbdd91405c6f584ea05b82a9438e3646750ae59..7487f5f4d6abaf3f4fa666ac9e0d8299b0ec2581 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -1860,6 +1860,10 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
             serverLevel.tickTimes10s.add(this.tickCount, tickDuration);
             serverLevel.tickTimes60s.add(this.tickCount, tickDuration);
             // DivineMC end - MSPT Tracking for each world
+        // DivineMC start - Catch update suppressors
+        } catch (org.bxteam.divinemc.util.exception.UpdateSuppressorException e) {
+            LOGGER.info(e.getMessage());
+        // DivineMC end - Catch update suppressors
         } catch (Throwable levelTickingException) {
             CrashReport crashReport = CrashReport.forThrowable(levelTickingException, "Exception ticking world");
             serverLevel.fillReportDetails(crashReport);
diff --git a/net/minecraft/world/level/block/ShulkerBoxBlock.java b/net/minecraft/world/level/block/ShulkerBoxBlock.java
index 0171868d8f28bd51d7d5b1ceef65c1447415d3f8..c6ce7c43ab1e6c92ac986dbf3b4566a219d4d24d 100644
--- a/net/minecraft/world/level/block/ShulkerBoxBlock.java
+++ b/net/minecraft/world/level/block/ShulkerBoxBlock.java
@@ -181,7 +181,17 @@ public class ShulkerBoxBlock extends BaseEntityBlock {
 
     @Override
     protected int getAnalogOutputSignal(final BlockState state, final Level level, final BlockPos pos, final Direction direction) {
-        return AbstractContainerMenu.getRedstoneSignalFromBlockEntity(level.getBlockEntity(pos));
+        // DivineMC start - Catch update suppressors
+        try {
+            return AbstractContainerMenu.getRedstoneSignalFromBlockEntity(level.getBlockEntity(pos));
+        } catch (ClassCastException ex) {
+            if (org.bxteam.divinemc.config.DivineConfig.FixesCategory.updateSuppressionCrashFix) {
+                throw new org.bxteam.divinemc.util.exception.UpdateSuppressorException(pos, this);
+            } else {
+                throw ex;
+            }
+        }
+        // DivineMC end - Catch update suppressors
     }
 
     public @Nullable DyeColor getColor() {
diff --git a/net/minecraft/world/level/redstone/NeighborUpdater.java b/net/minecraft/world/level/redstone/NeighborUpdater.java
index 72ef04091d1c850de5c15f0d9be3f2097a44ca7a..d9930ef56ef251a79787a7778fa6e17be69cd574 100644
--- a/net/minecraft/world/level/redstone/NeighborUpdater.java
+++ b/net/minecraft/world/level/redstone/NeighborUpdater.java
@@ -84,9 +84,19 @@ public interface NeighborUpdater {
             state.handleNeighborChanged(level, pos, changedBlock, orientation, movedByPiston);
             // Spigot start
         } catch (StackOverflowError ex) {
+            // DivineMC start - Catch update suppressors
+            if (org.bxteam.divinemc.config.DivineConfig.FixesCategory.updateSuppressionCrashFix) {
+                throw new org.bxteam.divinemc.util.exception.UpdateSuppressorException(pos, changedBlock);
+            }
+            // DivineMC end - Catch update suppressors
             level.lastPhysicsProblem = pos.immutable();
             // Spigot end
         } catch (Throwable t) {
+            // DivineMC start - Catch update suppressors
+            if (org.bxteam.divinemc.config.DivineConfig.FixesCategory.updateSuppressionCrashFix) {
+                throw new org.bxteam.divinemc.util.exception.UpdateSuppressorException(pos, changedBlock);
+            }
+            // DivineMC end - Catch update suppressors
             CrashReport report = CrashReport.forThrowable(t, "Exception while updating neighbours");
             CrashReportCategory category = report.addCategory("Block being updated");
             category.setDetail(
