From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NONPLAYT <76615486+NONPLAYT@users.noreply.github.com>
Date: Thu, 5 Jun 2025 17:40:39 +0300
Subject: [PATCH] temporary compile fix


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index e5c96be9970fe1f714ae3989145993e0ecc29c4f..289ad68425eb7f3f95fed9546699ff68d5f75290 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -2991,7 +2991,7 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
                         if (hand == InteractionHand.OFF_HAND && (level().purpurConfig.villagerCanBeLeashed || level().purpurConfig.wanderingTraderCanBeLeashed) && this instanceof net.minecraft.world.entity.npc.AbstractVillager) return InteractionResult.CONSUME; // Purpur - Allow leashing villagers
                         // Paper start - EntityUnleashEvent
                         if (!org.bukkit.craftbukkit.event.CraftEventFactory.handlePlayerUnleashEntityEvent(
-                            leashable2, player, hand, !player.hasInfiniteMaterials()
+                            leashable2, player, hand, !player.hasInfiniteMaterials(), true
                         )) {
                             ((ServerPlayer) player).connection.send(new net.minecraft.network.protocol.game.ClientboundSetEntityLinkPacket(this, leashable2.getLeashHolder()));
                             return InteractionResult.PASS;
@@ -3011,7 +3011,7 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
                         if (leashable2.isLeashed()) {
                             // Paper start - EntityUnleashEvent
                             if (!org.bukkit.craftbukkit.event.CraftEventFactory.handlePlayerUnleashEntityEvent(
-                                leashable2, player, hand, true
+                                leashable2, player, hand, true, true
                             )) {
                                 ((ServerPlayer) player).connection.send(new net.minecraft.network.protocol.game.ClientboundSetEntityLinkPacket(this, leashable2.getLeashHolder()));
                                 return InteractionResult.PASS;
@@ -3068,6 +3068,7 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
                 this,
                 player,
                 interactionHand,
+                true,
                 true
             );
             // Paper end - EntityUnleashEvent
@@ -3081,6 +3082,7 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
                 leashable1,
                 player,
                 interactionHand,
+                true,
                 true
             );
             // leashable1.dropLeash(); // Paper - EntityUnleashEvent - moved into handlePlayerUnleashEntityEvent
