--- a/net/minecraft/server/level/ServerPlayer.java
+++ b/net/minecraft/server/level/ServerPlayer.java
@@ -3157,7 +_,7 @@
 
     @Override
     public void swingAndResetAttackStrength(final InteractionHand hand, final SwingAnimation animation, final boolean sendToSwingingEntity) {
-        if (this.swing(hand, animation, sendToSwingingEntity)) {
+        if (this.swing(hand, animation, sendToSwingingEntity) && (hand == InteractionHand.MAIN_HAND || !org.bxteam.divinemc.config.DivineConfig.FixesCategory.fixMc118740)) { // DivineMC - Fix MC-118740
             this.resetAttackStrengthTicker();
         }
     }
