--- a/net/minecraft/server/level/ServerChunkCache.java
+++ b/net/minecraft/server/level/ServerChunkCache.java
@@ -595,8 +_,10 @@
             this.chunkMap.collectSpawningChunks(spawningChunks);
             profiler.popPush("shuffleSpawningChunks");
             // Paper start - chunk tick iteration optimisation
-            this.shuffleRandom.setSeed(this.level.getRandom().nextLong());
-            if (!this.level.paperConfig().entities.spawning.perPlayerMobSpawns) Util.shuffle(spawningChunks, this.level.getRandom()); // Paper - Optional per player mob spawns; do not need this when per-player is enabled
+            if (!this.level.paperConfig().entities.spawning.perPlayerMobSpawns) {
+                this.shuffleRandom.setSeed(this.level.getRandom().nextLong());
+                Util.shuffle(spawningChunks, this.level.getRandom()); // Paper - Optional per player mob spawns; do not need this when per-player is enabled
+            }
             // Paper end - chunk tick iteration optimisation
             profiler.popPush("tickSpawningChunks");
 
