--- a/net/minecraft/server/PlayerAdvancements.java
+++ b/net/minecraft/server/PlayerAdvancements.java
@@ -59,7 +_,7 @@
     private @Nullable AdvancementHolder lastSelectedTab;
     private boolean isFirstPacket = true;
     private final Codec<PlayerAdvancements.Data> codec;
-    private final Map<CriterionTrigger<?>, Map<PlayerAdvancements.TriggerInstanceKey, ? extends CriterionTriggerInstance>> activeTriggers = new IdentityHashMap<>();
+    private final Map<CriterionTrigger<?>, Map<PlayerAdvancements.TriggerInstanceKey, ? extends CriterionTriggerInstance>> activeTriggers = new it.unimi.dsi.fastutil.objects.Reference2ReferenceOpenHashMap<>(); // DivineMC - Use fastutil for criterion data
 
     public PlayerAdvancements(
         final DataFixer dataFixer, final PlayerList playerList, final ServerAdvancementManager manager, final Path playerSavePath, final ServerPlayer player
@@ -210,6 +_,11 @@
     }
 
     public boolean revoke(final AdvancementHolder advancement, final String criterion) {
+        // DivineMC start - Spectator don't get advancements
+        if (this.player.level().divineConfig.spectatorDontGetAdvancement && player.gameMode.getGameModeForPlayer() == net.minecraft.world.level.GameType.SPECTATOR) {
+            return false;
+        }
+        // DivineMC end - Spectator don't get advancements
         boolean result = false;
         AdvancementProgress progress = this.getOrStartProgress(advancement);
         boolean wasDone = progress.isDone();
