From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NONPLAYT <76615486+NONPLAYT@users.noreply.github.com>
Date: Sun, 29 Jun 2025 15:09:57 +0300
Subject: [PATCH] Smooth teleport API


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index efa472345ec74e21ca1cf05645b3ab39eda3caa4..4179317ac19a44d6f7ee6fdec4a5b2876e1daa08 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -1392,6 +1392,17 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         // Paper end - Teleportation API
     }
 
+    // DivineMC start - Smooth teleport API
+    @Override
+    public boolean teleportWithoutRespawn(Location location) {
+        ServerPlayer serverPlayer = getHandle();
+        serverPlayer.smoothWorldTeleport = true;
+        boolean teleportResult = teleport(location);
+        serverPlayer.smoothWorldTeleport = false;
+        return teleportResult;
+    }
+    // DivineMC end - Smooth teleport API
+
     @Override
     public boolean teleport(Location location, PlayerTeleportEvent.TeleportCause cause) {
         // Paper start - Teleport API
